src/org/intellij/erlang/completion/QuoteInsertHandler.java

```diff
@@ -52,26 +52,39 @@ public static void process(@NotNull Project project, @NotNull String name, @NotN
 
   public static void process(@NotNull Project project, @NotNull String name, @Nullable String moduleName, @NotNull InsertionContext context) {
     if (needQuotation(project, moduleName)) {
-      Editor editor = context.getEditor();
-      Document document = editor.getDocument();
-      context.commitDocument();
       int startOffset = context.getStartOffset();
-      document.insertString(startOffset, QUOTA);
-      document.insertString(startOffset + moduleName.length() + 1, QUOTA);
+      insertQuotesIfAbsent(context, startOffset, startOffset + moduleName.length());
     }
     if (needQuotation(project, name)) {
-      Editor editor = context.getEditor();
-      Document document = editor.getDocument();
-      context.commitDocument();
       int tailOffset = context.getTailOffset();
       int startOffset = moduleName == null ? context.getStartOffset() : tailOffset - name.length();
-      document.insertString(startOffset, QUOTA);
-      document.insertString(tailOffset + 1, QUOTA);
-      editor.getCaretModel().moveToOffset(tailOffset + 2);
+      insertQuotesIfAbsent(context, startOffset, tailOffset);
+      context.getEditor().getCaretModel().moveToOffset(tailOffset + 2);
       context.setTailOffset(tailOffset + 2);
     }
   }
 
+  private static void insertQuotesIfAbsent(@NotNull InsertionContext context, int startOffset, int tailOffset) {
+    Editor editor = context.getEditor();
+    Document document = editor.getDocument();
+    context.commitDocument();
+
+    if (!alreadyHasQuote(context, startOffset - 1)) {
+      document.insertString(startOffset, QUOTA);
+      context.commitDocument();
+      tailOffset++;
+    }
+    if (!alreadyHasQuote(context, tailOffset)) {
+      document.insertString(tailOffset, QUOTA);
+    }
+  }
+
+  private static boolean alreadyHasQuote(@NotNull InsertionContext context, int position) {
+    char[] file = context.getFile().textToCharArray();
+    if (file.length <= position) return false;
+    return QUOTA.equals(String.valueOf(context.getFile().textToCharArray()[position]));
```

Comment:
OMG. Are you serious? You copy a whole file text two times in order to check if it has a quote at an offset?